### PR TITLE
add pagination and OpenAPI

### PR DESCRIPTION
**Pull request by ihorromaniuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -29,21 +38,61 @@ public class BookController {
...
 
+    @Operation(summary = "Get all books with pagination")
+    @ApiResponse(
+            responseCode = "200",
+            description = "Found books with pagination"
+    )
     @GetMapping
-    public List<BookDto> getAllBooks() {
-        return bookService.findAll();
+    public List<BookDto> getAllBooks(@ParameterObject Pageable pageable) {
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java]: better return Page instead List

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -29,21 +38,61 @@ public class BookController {
...
+    @Operation(summary = "Get books with filters and pagination")
+    @ApiResponse(
+            responseCode = "200",
+            description = "Found books",
+            content = @Content(mediaType = "application/json")
+    )
     @PostMapping("/search")
-    public List<BookDto> getBooksBySpec(@RequestBody BookParamsDto params) {
-        return bookService.findAll(params);
+    public List<BookDto> getBooksBySpec(@RequestBody BookParamsDto params,
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java]: the same

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -29,21 +38,61 @@ public class BookController {
     private static final char PATH_SEPARATOR = '/';
     private final BookService bookService;
 
+    @Operation(summary = "Get all books with pagination")
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java:44]: Add Tag annotation above class

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -29,21 +38,61 @@ public class BookController {
...
+            @ApiResponse(
+                    responseCode = "400",
+                    description = "Validation failed",
+                    content = @Content(schema = @Schema(
+                            implementation = ValidationExceptionDto.class
+                    ))
+            )
+    })
     @PostMapping
     public ResponseEntity<?> createBook(@RequestBody @Valid CreateBookRequestDto requestDto,
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java]: Why do you need wrapping to ResponseEntity?

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s improve your solution ;)

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java (hunk 4 above)
Comment (ihorromaniuk) [reply src/main/java/core/basesyntax/bookstore/controller/BookController.java]: response status created + url header "Location" (easier done using response entity)

File: pom.xml [hunk 5]
```diff
@@ -77,6 +77,17 @@
...
 			<version>${liquibase.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+			<version>${openapi.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-starter-webflux-ui</artifactId>
```
Comment (Elena-Bruyako) [inline pom.xml]: do you need this dependency?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**